index.js

```diff
@@ -39,355 +32,291 @@ module.exports = Store;
+   * @return {Boolean} Returns true if `key` has
+   * @api public
+   */
+
+  has(key) {
+    assert.equal(typeof key, 'string', 'expected key to be a string');
+    return typeof get(this.data, key) !== 'undefined';
+  }
 
-/**
- * Clear in-memory cache.
- *
- * ```js
- * store.clear();
- * ```
- * @api public
- */
+  /**
+   * Returns `true` if the specified `key` exists.
+   *
+   * ```js
+   * store.set('a', 'b');
+   * store.set('b', false);
+   * store.set('c', null);
+   * store.set('d', true);
+   *
+   * store.hasOwn('a'); //=> true
+   * store.hasOwn('b'); //=> true
+   * store.hasOwn('c'); //=> true
+   * store.hasOwn('d'); //=> true
+   * store.hasOwn('foo'); //=> false
+   * ```
+   *
+   * @param  {String} `key`
+   * @return {Boolean} Returns true if `key` exists
+   * @api public
+   */
+
+  hasOwn(key) {
+    assert.equal(typeof key, 'string', 'expected key to be a string');
+    return hasOwn(this.data, key);
+  }
 
-Store.prototype.clear = function() {
-  this.cache = {};
-  this.data = {};
-  return this;
-};
+  /**
+   * Delete `keys` from the store, or delete the entire store
+   * if no keys are passed. A `del` event is also emitted for each key
+   * deleted.
+   *
+   * **Note that to delete the entire store you must pass `{force: true}`**
+   *
+   * ```js
+   * store.del();
+   *
+   * // to delete paths outside cwd
+   * store.del({force: true});
+   * ```
+   *
+   * @param {String|Array|Object} `keys` Keys to remove, or options.
+   * @param {Object} `options`
+   * @api public
+   */
+
+  del(key) {
+    if (!key) key = this.keys;
+    if (Array.isArray(key)) {
+      for (const k of key) this.del(k);
+      return this;
+    }
+    assert.equal(typeof key, 'string', 'expected key to be a string');
+    delete this.data[key];
+    this.emit('del', key);
+    this.save();
+    return this;
+  }
 
-/**
- * Delete `keys` from the store, or delete the entire store
- * if no keys are passed. A `del` event is also emitted for each key
- * deleted.
- *
- * **Note that to delete the entire store you must pass `{force: true}`**
- *
- * ```js
- * store.del();
- *
- * // to delete paths outside cwd
- * store.del({force: true});
- * ```
- *
- * @param {String|Array|Object} `keys` Keys to remove, or options.
- * @param {Object} `options`
- * @api public
- */
+  /**
+   * Stringify the store
+   */
 
-Store.prototype.del = function(keys, options) {
-  var isArray = Array.isArray(keys);
-  if (typeof keys === 'string' || isArray) {
-    keys = utils.arrayify(keys);
-  } else {
-    options = keys;
-    keys = null;
+  json(replacer = null, space = this.indent) {
+    return JSON.stringify(this.data, replacer, space);
   }
 
-  options = options || {};
-
-  if (keys) {
-    keys.forEach(function(key) {
-      proto.del.call(this, key);
-    }.bind(this));
-    this.save();
+  /**
+   * Persist the store to the file system.
+   *
+   * ```js
+   * store.save();
+   * ```
+   * @api public
+   */
+
+  save() {
```

Comment:
I wonder if there should be a "debounce" implementation here since when setting an object directly could cause `save` to be called multiple times very quickly and may cause issues with file system interaction. Idea of what this might look like:

```js
save() {
  if (this.timeout) {
    clearTimeout(this.timeout);
    this.timeout = null;
  }
  this.timeout = setTimeout(() => {
    this.timeout = null;
    mkdirSync(path.dirname(this.path), this.options.mkdir);
    fs.writeFileSync(this.path, this.json(), { mode: 0o0600 });
  }, 20);
}
```

Reply:
or we can do atomic writes using streams, maybe take a different approach.